client/src/tpu_client.rs

```diff
@@ -85,39 +87,33 @@ impl TpuClient {
     /// size
     pub fn send_transaction(&self, transaction: &Transaction) -> bool {
         let wire_transaction = serialize(transaction).expect("serialization should succeed");
-        self.send_wire_transaction(&wire_transaction)
+        self.send_wire_transaction(wire_transaction)
     }
 
     /// Send a wire transaction to the current and upcoming leader TPUs according to fanout size
-    pub fn send_wire_transaction(&self, wire_transaction: &[u8]) -> bool {
+    pub fn send_wire_transaction(&self, wire_transaction: Vec<u8>) -> bool {
         self.try_send_wire_transaction(wire_transaction).is_ok()
     }
 
     /// Serialize and send transaction to the current and upcoming leader TPUs according to fanout
     /// size
     /// Returns the last error if all sends fail
-    pub fn try_send_transaction(
-        &self,
-        transaction: &Transaction,
-    ) -> std::result::Result<(), std::io::Error> {
+    pub fn try_send_transaction(&self, transaction: &Transaction) -> TransportResult<()> {
         let wire_transaction = serialize(transaction).expect("serialization should succeed");
-        self.try_send_wire_transaction(&wire_transaction)
+        self.try_send_wire_transaction(wire_transaction)
     }
 
     /// Send a wire transaction to the current and upcoming leader TPUs according to fanout size
     /// Returns the last error if all sends fail
-    fn try_send_wire_transaction(
-        &self,
-        wire_transaction: &[u8],
-    ) -> std::result::Result<(), std::io::Error> {
-        let mut last_error: Option<std::io::Error> = None;
+    fn try_send_wire_transaction(&self, wire_transaction: Vec<u8>) -> TransportResult<()> {
+        let mut last_error: Option<TransportError> = None;
         let mut some_success = false;
 
         for tpu_address in self
             .leader_tpu_service
             .leader_tpu_sockets(self.fanout_slots)
         {
-            let result = self.send_socket.send_to(wire_transaction, tpu_address);
+            let result = send_wire_transaction_async(wire_transaction.clone(), &tpu_address);
```

Comment:
an `_async` version of this function (and `sync fn send_wire_transaction`) would allow us to do the address iteration inside the runtime and avoid the cloning, right?

Reply:
Sorry, can you be more explicit? I think what you're asking for would require shaking up the TpuConnection trait, but I'm not yet sure I follow.